src/StmtToViz.cpp

```diff
@@ -0,0 +1,2900 @@
+    }
+
+    void visit(const Prefetch *op) override {
+        print_opening_tag("div", "Prefetch");
+
+        // Print cost buttons
+        print_cost_buttons(op);
+
+        // Print prefetch
+        print_html_element("span", "matched keyword", "prefetch ");
+        print_variable(op->name);
+        print_html_element("span", "matched", "(");
+        for (size_t i = 0; i < op->bounds.size(); i++) {
+            print_html_element("span", "matched", "[");
+            print(op->bounds[i].min);
+            print_html_element("span", "matched", ",");
+            print(op->bounds[i].extent);
+            print_html_element("span", "matched", "]");
+            if (i < op->bounds.size() - 1) {
+                print_html_element("span", "matched", ", ");
+            }
+        }
+        print_html_element("span", "matched", ")");
+
+        // Print condition
+        if (!is_const_one(op->condition)) {
+            print_html_element("span", "keyword", " if ");
+            print(op->condition);
+        }
+
+        // Print prefetch body
+        print_opening_tag("div", "indent PrefetchBody");
+        print(op->body);
+        print_closing_tag("div");
+
+        print_closing_tag("div");
+    }
+
+    void visit(const Atomic *op) override {
+        // Give this node a unique id
+        int id = gen_unique_id();
+
+        // Start a dive to hold code for this atomic
+        print_opening_tag("div", "Atomic");
+
+        // Generate the show hide icon/text buttons
+        print_toggle_anchor_opening_tag(id);
+
+        // -- print icon
+        print_show_hide_icon(id);
+
+        // -- print text
+        print_html_element("span", "matched keyword", "atomic");
+        if (!op->mutex_name.empty()) {
+            print_html_element("span", "matched", "(");
+            print_html_element("span", "Symbol", op->mutex_name);
+            print_html_element("span", "matched", ")");
+        }
+
+        print_toggle_anchor_closing_tag();
+
+        // Open code block to hold atomic body
+        print_html_element("span", "matched", " {");
+
+        // Open indented div to hold atomic code
+        print_opening_tag("div", "indent AtomicBody", id);
+
+        // Print atomic body
+        print(op->body);
+
+        // Close indented div holding body code
+        print_closing_tag("div");
+
+        // Close code block holding fork body
+        print_html_element("span", "matched ClosingBrace cb-" + std::to_string(id), "}");
+
+        // Close div holding this atomic
+        print_closing_tag("div");
+    }
+};
+
+/************** HTMLVisualizationPrinter Class ***************/
+// Visualizes the IR in HTML. The visualization is essentially
+// an abstracted version of the code, highlighting the higher
+// level execution pipeline along with key properties of the
+// execution performed at each stage.
+class HTMLVisualizationPrinter : public IRVisitor {
+public:
+    HTMLVisualizationPrinter(std::ofstream &os)
```

Comment:
style nit, here and elsewhere: constructors with a single argument should ~always be marked as `explicit`. (In the unusual cases where implicit construction is desired, the ctor should have a comment like `/* not-explicit */` in front of it to indicate this is deliberate.)